scriptworker/test/test_utils.py

```diff
@@ -307,3 +350,37 @@ def test_load_json(string, is_path, exception, raises, result):
             utils.load_json(string, is_path=is_path, exception=exception)
     else:
         assert result == utils.load_json(string, is_path=is_path, exception=exception)
+
+
+# match_url_regex {{{1
+def test_match_url_regex():
+    rules =  ({
+        'schemes': ['bad_scheme'],
+    }, {
+        'schemes': ['https'],
+        'netlocs': ['bad_netloc'],
+    }, {
+        'schemes': ['https'],
+        'netlocs': ['hg.mozilla.org'],
+        'path_regexes': [
+            "^bad_regex$",
+        ],
+    }, {
+        'schemes': ['https'],
+        'netlocs': ['hg.mozilla.org'],
+        'path_regexes': [
+            "^.*$",
+            "^/(?P<path>mozilla-(central|unified))(/|$)",
+        ],
+    })
+
+    def cb(m):
+        import logging
+        log = logging.getLogger()
+        log.error(m)
+        path_info = m.groupdict()
+        if 'path' in path_info:
+            return path_info['path']
+
+    assert utils.match_url_regex(rules, "https://hg.mozilla.org/mozilla-central", cb) == "mozilla-central"
```

Comment:
I'm not sure to understand this test. Do we verify that: even though rules contain bad input, we manage to match the URL because one of the rules is valid?